src/core/ircmessage.cpp

```diff
@@ -26,6 +26,8 @@
   SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
 
+#include <functional>
```

Comment:
move this as last include. stdlib, qt, or other 3rd party headers should not be included before libcommuni headers.